### PR TITLE
feat: add CAS support for helpers.getProof

### LINKED ISSUE #115: simplify proof generation by using period store
## Bounty

the [period store](https://github.com/leapdao/leap-node/issues/289) is meant to reduce the complexity to create proofs for transaction inclusion. 

## Scope
- change current `getProof()` to use period store
- do not require `slotId` or `validatorAddr` as inputs for proof generation

## Deliverables
- code
- test

## Gain for the project
- run multiple validators

## Roles
bounty gardener: name / share
bounty worker: name / share
bounty reviewer: name / share

### PR DESCRIPTION
**Pull request by troggy**
- added `plasma_getPeriodByBlockHeight` JSON RPC extension ([docs](https://github.com/leapdao/leapdao-docs/pull/39))
- improved `getProof` helper so it fetches period data automatically:

```ts
helpers.getProof(plasma: ExtendedWeb3 | LeapEthers, tx: LeapTransaction)
```

Resolves #115

### CODE REVIEW

File: lib/period.js [hunk 1]
```diff
@@ -57,6 +57,7 @@ export default class Period {
   }
 
   setValidatorData(slotId, ownerAddr, casBitmap) {
+    console.log(slotId, ownerAddr, casBitmap, toBuffer(ownerAddr))
```
Comment (johannbarbie) [inline lib/period.js]: still needed?

Comment (troggy) [reply lib/period.js]: ops

File: lib/helpers.js [hunk 2]
```diff
@@ -240,11 +251,20 @@ export function getYoungestInputTx(plasma, tx) {
...
-    period.setValidatorData(slotId, validatorAddr);
-    return period.proof(Tx.fromRaw(tx.raw));
-  });
+export function getProof(plasma, tx) {
+  return Period.periodForTx(plasma, tx)
+    .then((period) => {
+      const periodData = plasma.getPeriodByBlockHeight(tx.blockNumber);
+      return Promise.all([period, periodData]);
+    })
+    .then(([period, periodData]) => {
```
Comment (sunify) [inline lib/helpers.js]: Why not

```es6
return Promise.all([
  Period.periodForTx(plasma, tx),
  plasma.getPeriodByBlockHeight(tx.blockNumber)
]).then(([period, periodData]) => {
  ...
});
```

?

Comment (troggy) [reply lib/helpers.js]: 🤦‍♂️

Thanks!

**Pull request merged**